shenyu-admin/src/main/java/org/apache/shenyu/admin/shiro/config/ShiroRealm.java

```diff
@@ -64,19 +71,32 @@ protected AuthorizationInfo doGetAuthorizationInfo(final PrincipalCollection pri
 
     @Override
     protected AuthenticationInfo doGetAuthenticationInfo(final AuthenticationToken authenticationToken) {
-        StatelessToken token = (StatelessToken) authenticationToken;
-        if (token == null || token.getToken() == null) {
+        String token = (String) authenticationToken.getCredentials();
+        if (StringUtils.isEmpty(token)) {
             return null;
         }
 
-        String userName = JwtUtils.getIssuer(token.getToken());
+        UserInfo userInfo = checkUserTokenIsValid(token);
 
-        token.setUserName(userName);
+        return new SimpleAuthenticationInfo(userInfo, token, this.getName());
+    }
 
-        if (JwtUtils.getUserId() == null) {
-            JwtUtils.setUserId(token.getToken());
+    /**
+     * check token valid.
+     *
+     * @param token user token
+     * @return userInfo {@link UserInfo}
+     */
+    private UserInfo checkUserTokenIsValid(final String token) {
```

Comment:
Does `getUserInfoByToken/getUserInfo` would be better? I suggest throws `AuthenticationException` directly If the user can't not be found.